Convert to use new headers library

janderson2 opened the pull request:
Update header setting logic to use the new headers library, replacing
usage of the deprecated `go-ns/common` header functions.

janderson2 committed:
Convert to use new headers library

Update header setting logic to use the new headers library, replacing
usage of the deprecated `go-ns/common` header functions.

nshumoogum commented on client.go:
```diff
@@ -12,6 +12,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ONSdigital/dp-api-clients-go/headers"
```
I know we can use dependency manager to handle which version of an import we should be using but is this not bad practice to have libraries which depend on each other? Resulting in cyclic dependency

daiLlew replied:
That is a good point. Being completely objective I don't think the `dp-rchttp` client should know about anything about headers. 
The role of the client is to send/retry failed request and that should be all it does. Having specific header knowledge here blurs the line of responsibility, breaks encapsulation and creates a coupled code. It should be up to the caller to prepare the request the client should only be concerned with sending it and retrying if it fails.

> Do only one thing but do it well.

janderson2 replied:
While I agree that this library shouldn't have any knowledge of headers, that is something we have been moving towards with the repo over time, but is a separate piece of work requiring far too much refactoring of all of our libraries and apps to be feasible at this time.

Furthermore, there isn't a cyclic dependency here since cyclic dependencies are between packages not modules.  It is certainly a code smell, but as this PR is not adding the cyclic dependency between modules (just moving it from `go-ns` to `dp-api-clients-go` it does not seem reasonable to block this work over.

janderson2 committed:
Remove temporary code setting Florence token

All API clients in `dp-api-clients-go` should pass the appropriate auth
headers now.

janderson2 committed:
Update to latest go-ns

janderson2 committed:
Remove dependency on `dp-api-clients-go/headers`

Remove the dependency on `dp-api-clients-go/headers` in order to
simplify the overall dependency tree of our libraries.  Although this
means the header key is declared in two places, it should only be
temporary until we move to a more standard approach of trace
propagation.
